engine/src/conversion/analysis/fun/mod.rs

```diff
@@ -80,10 +80,11 @@ pub(crate) enum ReceiverMutability {
     Mutable,
 }
 
-#[derive(Clone)]
+#[derive(Clone, Debug)]
 pub(crate) enum MethodKind {
     Normal(ReceiverMutability),
     Constructor,
+    DefaultConstructor,
```

Comment:
`MethodKind::Constructor` is referred to in lots of places, and I can see you've added corresponding `DefaultConstructor` variants. But future maintainers (me, for example) may not be so careful and diligent. To avoid these things getting out of sync, please would you remove `DefaultConstructor` and instead change `Constructor` to `Constructor { is_default: bool }` or `Constructor(ConstructorDefaultness)` or something along those lines.

Reply:
Done